### PR TITLE
[DNM] store/tikv: add FastCommit (1PC)

### PR DESCRIPTION
**Pull request by disksing**

### CODE REVIEW

File: store/tikv/txn_committer.go [hunk 1]
```diff
@@ -285,6 +297,82 @@ func (c *txnCommitter) Commit() error {
 	return nil
 }
 
+func (c *txnCommitter) fastCommit() (bool, error) {
+	// Check if all keys are located on the same.
```
Comment (shenli) [inline store/tikv/txn_committer.go]: on/in the same region

File: store/tikv/mock-tikv/mvcc.go [hunk 2]
```diff
@@ -141,6 +141,25 @@ func (e *mvccEntry) Commit(startTS, commitTS uint64) error {
...
+func (e *mvccEntry) FastCommit(mutation *kvrpcpb.Mutation, startTS, commitTS uint64) error {
+	if len(e.values) > 0 {
+		if e.values[0].commitTS >= startTS {
+			return ErrRetryable("write conflict")
+		}
+	}
+	if e.lock != nil {
+		return e.lockErr()
+	}
+	if mutation.GetOp() != kvrpcpb.Op_Lock {
```
Comment (shenli) [inline store/tikv/mock-tikv/mvcc.go]: I think we need a TODO here. It may not satisfy SelectForUpdate in some scenarios.

Comment (disksing) [inline store/tikv/mock-tikv/mvcc.go]: This is how our txn model works currently, which differs from MySQL.

**Comment by disksing** [comment]
@shenli @ngaut @c4pt0r PTAL

**Comment by shenli** [comment]
LGTM

**Comment by disksing** [comment]
Not able to find a simple solution for the isolation problem https://github.com/pingcap/tidb/pull/1261 . Close it for now.

**Pull request closed without merging**